datafusion/src/physical_plan/limit.rs

```diff
@@ -232,6 +249,24 @@ impl ExecutionPlan for LocalLimitExec {
         self.input.output_partitioning()
     }
 
+    fn relies_on_input_order(&self) -> bool {
+        self.input.output_ordering().is_some()
```

Comment:
This feels like an optimization that really belongs in the Repartition optimizer, namely that if the children of a plan don't have a sort order, you can freely repartition them even if the parent `relies_on_input_order`.